### PR TITLE
Expose onBlur for Field

### PR DESCRIPTION
**Pull request by kjkurtz**
The idea is that no matter the type of Field, we can use `onBlur` to get the result after the user stops interacting with that field (either leaves the text box, checks the box, etc.). This way we can trigger things like storing that field once they are theoretically done editing it for now.

### CODE REVIEW

File: packages/ui/src/Field.tsx [hunk 1]
```diff
@@ -132,7 +140,9 @@ export const Field = ({
           name={name}
           switched={Boolean(value)}
           testID={testID}
-          onChange={(result) => handleSwitchChange(result)}
+          onChange={(result) => {
+            handleSwitchChange(result);
```
Comment (joshgachnang) [inline packages/ui/src/Field.tsx:144]: Do we want to onBlur here too? It's kinda a weird line to have select field do it but not switch, but I could go any of the 3 ways (both, none, only Select)

File: packages/ui/src/Field.tsx [hunk 2]
```diff
@@ -243,7 +258,12 @@ export const Field = ({
           options={options}
           placeholder={placeholder}
           value={value}
-          onChange={onChange}
+          onChange={(val) => {
+            {
```
Comment (joshgachnang) [inline packages/ui/src/Field.tsx:262]: Extra curlies?

Comment (kjkurtz) [reply packages/ui/src/Field.tsx]: They were extra!

File: packages/ui/src/Field.tsx (hunk 1 above)
Comment (kjkurtz) [reply packages/ui/src/Field.tsx:144]: This actually does call `onBlur` already if you look at the `handleSwitchChange` function.

Comment (joshgachnang) [reply packages/ui/src/Field.tsx:144]: Ah duh!

**Review by joshgachnang — APPROVED** [review]

**Pull request merged**